src/components/States.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import { API_BASE_URL, API_KEY } from '../api';
+import Search from './Search';
+
+const States = () => {
+  const [states, setStates] = useState([]);
+  const [filteredStates, setFilteredStates] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  const fetchStates = async () => {
+    try {
+      setError(null);
+      setLoading(true);
+      const response = await fetch(
+        `${API_BASE_URL}/states?country=Canada&key=${API_KEY}`,
+      );
+      if (!response.ok) {
+        throw new Error('Failed to fetch data');
+      }
+      const data = await response.json();
+      setStates(data.data);
+      setFilteredStates(data.data);
+      setLoading(false);
+    } catch (error) {
+      setError(error);
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchStates();
+  }, []);
+
+  const handleSearch = (query) => {
+    const filtered = states.filter(
+      (state) => state.state.toLowerCase().includes(query.toLowerCase()),
+    );
+    setFilteredStates(filtered);
+  };
+
+  const renderStates = () => {
+    if (loading) {
+      return <p className="loading">Loading...</p>;
+    }
+    if (error) {
+      return (
+        <div className="error">
+          <p>Error fetching data. Please try again.</p>
+          <button type="button" onClick={fetchStates}>
+            Retry
+          </button>
+        </div>
+      );
+    }
+    return (
+      <ul className="items">
+        {filteredStates.map((state) => (
+          <li key={state.state} className="flex">
+            <Link to={`/${state.state}`} className="item-link">
+              {state.state}
+            </Link>
+          </li>
+        ))}
+      </ul>
+    );
+  };
+
+  return (
+    <div>
+      <nav className="navbar flex">
+        <h2>Canadian States</h2>
+        <Search onSearch={handleSearch} />
+      </nav>
+      {renderStates()}
+    </div>
+  );
+};
```

Comment:
- [ ] As required for the project to follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) including `Typography`(`font face`, `size`, `weight`,...), `Layout`(`composition` and `space between elements`,...), I think it would be better if you improve the alignment of your element text, cards to have some different colors(light and dark) as on the original design. I can also suggest you add the missing arrow in each card element, as this can also help users know if they are clickable.

| Template Design | Your Design |
| ---------------------- | ------------------- |
| 
![Screenshot from 2023-09-14 04-09-50](https://github.com/tajemouti/weather-pollution-metrics/assets/83042742/6b66c805-adec-4709-a0c4-895b839188b8) | ![Screenshot from 2023-09-14 04-11-00](https://github.com/tajemouti/weather-pollution-metrics/assets/83042742/19cf7ac1-7c6f-4368-9154-4f7a416d02d7) |